main.go

```diff
@@ -14,96 +11,114 @@ import (
 	"github.com/pkg/errors"
 )
 
-const ideasListID = "5b613db79ea6a782ac173a48"
-const scheduledListID = "5b613dbfd923da512f85263b"
-
-var client trelloClientAdapter
+const (
+	ideasListID     = "5b613db79ea6a782ac173a48"
+	scheduledListID = "5b613dbfd923da512f85263b"
+)
 
 func main() {
+	srv := newServer()
+	srv.start()
+	<-srv.done
+}
 
-	token := os.Getenv("SLACK_TOKEN")
-	api := slack.New(token)
-	api.SetDebug(true)
-
-	rtm := api.NewRTM()
-	// goroutine, async exec
-	go rtm.ManageConnection()
+type server struct {
+	trello TrelloClient
+	slack  SlackRTMInterface
+	done   chan struct{}
+}
 
-Loop:
-	for {
-		select {
-		// channel operator, await the async goroutine
-		case msg := <-rtm.IncomingEvents:
-			fmt.Print("Event Received: ")
-			switch ev := msg.Data.(type) {
-			case *slack.ConnectedEvent:
-				fmt.Println("Connection counter:", ev.ConnectionCount)
+func newServer() server {
+	appKey := os.Getenv("TRELLO_KEY")
+	trello := os.Getenv("TRELLO_TOKEN")
+	slack := os.Getenv("SLACK_TOKEN")
 
-			case *slack.MessageEvent:
-				fmt.Printf("Message: %v\n", ev)
-				info := rtm.GetInfo()
-				prefix := fmt.Sprintf("<@%s> ", info.User.ID)
+	return server{
+		trello: newTrelloClient(appKey, trello),
+		slack:  newSlackRTM(slack),
+		done:   make(chan struct{}),
+	}
+}
 
-				if ev.User != info.User.ID && strings.HasPrefix(ev.Text, prefix) {
-					response, err := execute(ev.Text, prefix, getListItems, addIdea, getHelp, getDadJoke)
+func (s *server) start() {
+	fmt.Println("Starting")
+
+	go func(done chan struct{}, s *server) {
+		for {
+			select {
+			// channel operator, await the async goroutine
+			case <-s.done:
```

Comment:
thought this was sort of like an await and the control would stay here until there was something on the done channel, think i've got to do some more reading on channels :)

Reply:
Yep it will wait until it gets a message or a close. However because it's part of a select, it will wait for either of the channels to do something.